src/mqtt_bridge/app.py

```diff
@@ -23,6 +24,44 @@ def config(binder):
         binder.bind('mqtt_private_path_extractor', private_path_extractor)
     return config
 
+# Method to read the param file again to configure
+# the mqtt client without stopping the node.
+# Can't use mqtt_bridge_node again because dependency injection
+# doesn't work.
+def reconnect():
+    
+    params = rospy.get_param("~", {})
+    mqtt_params = params.pop("mqtt", {})
+    conn_params = mqtt_params.pop("connection")
+    mqtt_private_path = mqtt_params.pop("private_path", "")
+    bridge_params = params.get("bridge", [])
+
+    # create mqtt client
+    mqtt_client_factory_name = rospy.get_param(
+        "~mqtt_client_factory", ".mqtt_client:default_mqtt_client_factory")
+
+    mqtt_client_factory = lookup_object(mqtt_client_factory_name)
+    mqtt_client = mqtt_client_factory(mqtt_params)
+
+    # load serializer and deserializer
+    serializer = params.get('serializer', 'json:dumps')
+    deserializer = params.get('deserializer', 'json:loads')
+
+    # configure and connect to MQTT broker
+    mqtt_client.on_connect = _on_connect
+    mqtt_client.on_disconnect = _on_disconnect
+    mqtt_client.connect(**conn_params)
+
+    # configure bridges
+    bridges = []
+    for bridge_args in bridge_params:
+        bridges.append(create_bridge(**bridge_args))
+
+    global counter
+    counter = 0
+
+    # start MQTT loop
+    mqtt_client.loop_start()
```

Comment:
Thanks @ggari-robotnik particularly for an attempt to fix the reconnection problems. It's not the most elegant solution yet, but in particular also @gpdas will find this helpful. Have you got a good test case for the reconnection implementation? Shouldn't we probably just move a lot of the login into the `_on_connect` method instead of implementing a specific new reconnect? I feel also my old implementation was rubbish, and the entire bridge construction should happen in `_on_connect`, and also ROS services and topics should be deregistered in `_on_disconnect`.

Reply:
Thinking of it further, it may be enough to re-issue to restart the mqtt loop, but needs testing.